CppCoreGuidelines.md

```diff
@@ -13791,168 +13793,174 @@ Performance rule summary:
 
 ##### Reason
 
-Simple code can be very fast. Optimizers sometimes do marvels with simple code
+Time is secondary metric.  Large time values are an effect of either bad
+orchestration, large resource consumption (read too much, use an O(N^2)
+algorithm instead of an O(N) alorigthm), or inefficient use of a resource (i.e.
+asking it to do something it's bad at, such as seeking all over, or dealing with
+non-local memory access).  Typically secondary metrics like time are not
+probative -- "the code is slow, boo hoo" -- primary metrics like "how many
+instructions ran", "what fraction of branches were predicted", "how many bytes
+were read" can be very probative.  Consumption metrics are very helpful to
+understanding what is going wrong and how you might fix it.
 
-##### Example, good
+Likewise, if poor orchestration is the problem (work spread across too many
+threads, too few threads, too much work issued at once, etc.) mapping out your
+orchestration will be very helpful.  The total time is likely not actionable
+directly.  Insights rarely come from secondary metrics.  Typically they tell you
+where to look further at best, you then need primary metrics to make
+improvements.
 
-    // clear expression of intent, fast execution
+Keep in mind that you should understand how your system performs according to
+your goals, not just its latency.  A system might be plenty fast enough and
+still use more memory than you can afford. Or more power than you can afford.
+Or be too big to reasonably download.  If you are paying for CPU cycles you
+are likely to be very excited about reducing cycles by 50% even if your customers
+already think the system is fast enough.  Cycles are cost.
 
-    vector<uint8_t> v(100000);
+Whatever your system may be, referring back to your needs and your customer
+needs will tell you where you should be investing and which investments are
+worthwhile. At some point, the potential benefits are not worth the opportunity
+cost of doing them.  An extra 10% reduction of download size maybe isn't worth
+the engineering time because you've reached a size that is small enough that
+your customers no longer notice the difference.  But if your needs also include
+egress costs for those downloads then maybe that same 10% is worth it.  The very
+same potential improvement might be a great a idea or a silly one depending on
+your needs.
 
-    for (auto& c : v)
-        c = ~c;
+### <a name="Rper-simple"></a>Per.4: Simpler is generally better
 
-##### Example, bad
+Assuming you have done the basics of making sure that your overall design is approximately right,
+such as you are using O(N) if it is available rather than O(N^2) or something like that then you'll
+find that *simplifying* code and data is the best way to make things faster.  This means:
 
-    // intended to be faster, but is often slower
+* don't create complex inheritance graphs
+* don't use complicate dispatch logic if simple logic will do
```

Comment:
```suggestion
* don't use complicate dispatch logic if simple logic will do
```

Reply:
changed to complicated.

Reply:
Oops, that's what I was trying to suggest but apparently failed to do! I shouldn't do reviews on my phone before breakfast.